chain/neutrino.go

```diff
@@ -199,6 +199,11 @@ func (s *NeutrinoClient) FilterBlocks(
 	// the filter returns a positive match, the full block is then requested
 	// and scanned for addresses using the block filterer.
 	for i, blk := range req.Blocks {
+		// TODO(wilmer): Investigate why polling it still necessary
```

Comment:
isn't this because the filter headers can lag behind the block headers?

Reply:
I'm not sure since the header chains should be caught up to tip before we attempt recovery:
https://github.com/btcsuite/btcwallet/blob/60299130c647f27394e818079e2a0705e6aa8e60/wallet/wallet.go#L351